### PR TITLE
ENG-1651 / Add getTransactions method

### PR DESCRIPTION
**Pull request by DiegoCerdaC**
### **PR Description**

Jira Ticket: [ENG-1651](https://sphereone.atlassian.net/browse/ENG-1651)

This PR includes some changes in the SDK to have a method to get the user transactions.

### **ChangeLogs**

- New `utils.ts` file containing a method to decode a payload in JWT format.
- New private method that uses the server endpoint to get the user transactions.
- New public method to fetch user transactions, includes optional parameters to fetch:
  - only sent transactions.
  - only received transactions.
  - get a predetermined number of transactions.

### **Designs**

`No designs`

### **Additional Notes**

This PR is related to [another PR on server repository](https://github.com/SphereGlobal/server/pull/500)

[ENG-1651]: https://sphereone.atlassian.net/browse/ENG-1651?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by FajardoSatino — COMMENTED** [review]
Looks good for me! Just a minor comment, if is not possible do that changes we can leave and create a ticket for that

File: dist/index.d.ts [hunk 1]
```diff
@@ -32,6 +31,7 @@ declare class WebSDK implements iWebSDK {
     createCharge: (charge: ChargeReqBody) => Promise<any>;
     pay: ({ toAddress, chain, symbol, amount, tokenAddress }: Transaction) => Promise<any>;
     payCharge: (transactionId: string) => Promise<any>;
+    getTransactions: (quantity?: number, getReceived?: boolean, getSent?: boolean) => Promise<any>;
```
[the remark this answers is not in the document]
Comment (DiegoCerdaC) [reply dist/index.d.ts:34]: @FajardoSatino sure!

**Review by FajardoSatino — APPROVED** [review]
Looks good to me!

File: index.ts [hunk 2]
```diff
@@ -362,6 +375,37 @@ class WebSDK implements iWebSDK {
...
+  getTransactions = async (
+    quantity = 0,
+    getReceived = true,
+    getSent = true
+  ): Promise<Transaction[]> => {
+    const encoded = await this.#fetchTransactions();
+    const { transactions } = await decodeJWT(encoded);
+    let response = transactions;
+    if (!getReceived) response = transactions.filter((t: any) => t.receiverUid !== this.user?.uid);
+    if (!getSent) response = transactions.filter((t: any) => t.senderUid !== this.user?.uid);
```
Comment (matzapatasphereone) [inline index.ts:387]: can you add some comments in all this, what the function does, what are you doing w these filters, mostly as docs. Thanks

**Review by matzapatasphereone — APPROVED** [review]
One comment to add some minimum docs, but is fine

**Pull request merged**